### PR TITLE
Fix #1851 meta data added to logged csv files

### LINKED ISSUE #1851: Add meta data to identify instrument to logged csv files
**Actual Behaviour**
Currently there is no meta in the logged csv file regarding the instrument.

**Expected Behaviour**
Ad a row at top containing instrument name which can be used to identify the csv file . Also make logged csv file consistent across Android and Desktop app 

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by neel1998**
Fixes #1851 

**Changes**: Instrument name added in the fisrt row of logged csv. So while importing the logged file user doesn't have to select the instrument manually, App will automatically detect the instrument from the metadata

**Screenshot/s for the changes**: 

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

**APK for testing**: 
[new_csv.zip](https://github.com/fossasia/pslab-android/files/3363691/new_csv.zip)

### CODE REVIEW

**Comment by AakashMallik** [comment]
LGTM...

**Comment by ho-dor** [comment]
@neel1998 codacy is solvable.

**Comment by neel1998** [comment]
@mariobehling please review

File: app/src/main/java/io/pslab/fragment/ThermometerDataFragment.java [hunk 1]
```diff
@@ -437,6 +441,10 @@ private void writeLogToFile(long timestamp, float sensorReading) {
         if (getActivity() != null && thermoSensor.isRecording) {
             if (thermoSensor.writeHeaderToFile) {
                 thermoSensor.csvLogger.prepareLogFile();
+                SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+                String metaDataTime = sdf.format(System.currentTimeMillis());
+                String metaData = getResources().getString(R.string.thermometer) + "," + metaDataTime.split(" ")[0] + "," + metaDataTime.split(" ")[1];
+                thermoSensor.csvLogger.writeCSVFile(metaData);
```
Comment (CloudyPadmal) [inline app/src/main/java/io/pslab/fragment/ThermometerDataFragment.java]: I see these 4 lines everywhere. In Java we can create a method with this repetitive code and replace these 4 lines with one line method call and implement the function inside CSVLogger class.

File: app/src/main/java/io/pslab/fragment/ThermometerDataFragment.java [hunk 2]
```diff
@@ -333,6 +333,10 @@ public void saveGraph() {
             try {
                 csvFile.createNewFile();
                 PrintWriter out = new PrintWriter(new BufferedWriter(new FileWriter(csvFile, true)));
+                SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
```
Comment (CloudyPadmal) [inline app/src/main/java/io/pslab/fragment/ThermometerDataFragment.java]: I wonder why we are rewriting the whole code that's already implemented in `CSVLogger.java` file. `writeCSVFile(String data)` does the same thing. I can see this in many files in this PR

**Comment by neel1998** [comment]
@CloudyPadmal  this PR got messy. I am closing this and opening a new.

**Comment by CloudyPadmal** [comment]
So much for best practices and my reviews. :+1: 

Note: #1857 is linked

**Pull request closed without merging**